eva-server/src/test/java/uk/ac/ebi/eva/server/ws/ga4gh/beaconv2/GA4GHBeaconWSServerV2Test.java

```diff
@@ -0,0 +1,165 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.ac.ebi.eva.commons.beacon.models.BeaconAlleleRequest;
+import uk.ac.ebi.eva.commons.beacon.models.BeaconAlleleResponse;
+import uk.ac.ebi.eva.commons.beacon.models.Chromosome;
+import uk.ac.ebi.eva.commons.core.models.Region;
+import uk.ac.ebi.eva.commons.core.models.VariantType;
+import uk.ac.ebi.eva.commons.mongodb.entities.VariantMongo;
+import uk.ac.ebi.eva.commons.mongodb.filter.FilterBuilder;
+import uk.ac.ebi.eva.commons.mongodb.filter.VariantRepositoryFilter;
+import uk.ac.ebi.eva.commons.mongodb.services.VariantWithSamplesAndAnnotationsService;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.eq;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class GA4GHBeaconWSServerV2Test {
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @MockBean
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @Before
+    public void setup() throws Exception {
+        VariantMongo variantMongo = new VariantMongo(null, "X", 100470026, 100470026, 1, "G", "A");
+        List<VariantMongo> variantMongoList = Collections.singletonList(variantMongo);
+        Region startRange = new Region("X", 100470026L, 100470026L);
+        Region endRange = new Region("X", 100470026L, 100470026L);
+        List<VariantRepositoryFilter> variantRepositoryFilters = new FilterBuilder().getBeaconFilters("G", "A",
+                VariantType.SNV, Arrays.asList("PRJEB7218"));
+
+        Pageable pageable = new PageRequest(0, 1);
+        given(service.findByRegionAndOtherBeaconFilters(eq(startRange), eq(endRange), eq(variantRepositoryFilters),
+                eq(pageable))).willReturn(variantMongoList);
+        given(service.countByRegionAndOtherBeaconFilters(eq(startRange), eq(endRange), eq(variantRepositoryFilters)))
+                .willReturn(1L);
+    }
+
+    @Test
+    public void testForExisting() throws Exception {
+        BeaconAlleleRequest request = new BeaconAlleleRequest();
+        request.setReferenceName(Chromosome.X);
+        request.setAssemblyId("GRCh37");
+        request.setReferenceBases("G");
+        request.setStart(100470026L);
+        request.setEnd(100470026);
+        request.setAlternateBases("A");
+        request.setVariantType("SNV");
```

Comment:
Wouldn't the usual request contain either the alternate base OR the variant type?

Reply:
Yes, I have made it such that only one of them is used where the url is built :+1: 
Also, I think that both of them can be passed together as well(I didn't find such a rule)

Reply:
Wouldn't that be problematic from a usability point of view? How will users know which one is prioritized?

> Optional: either `alternateBases` or `variantType` is required.

This is literally what the spec says, so you're right about no rule stating that only one of them can be passed. But that means that at least one must be provided (the input must be validated accordingly), but if both are provided, they should be evaluated.

Reply:
Yeah, I am considering the case where both of them comes as well in the code :)

Reply:
Could you add a new test that checks how changing the type of a variant previously found with a combination of alt+type, returns no results?